test/database.yml

```diff
@@ -1,6 +1,6 @@
 sqlite:
   adapter: sqlite3
-  database: "file:memdb1?mode=memory&cache=shared"
+  database: ":memory:"
```

Comment:
I _think_ the intent of `mode=memory` is to use an in-memory SQLite database, but that doesn't seem to be working, so this is a functional change (at least, for me).

This isn't an essential part of this PR, but has a couple advantages:

* in-memory tests run faster
* there's no risk of the test suite exiting early and leaving a locked DB file around

Both of these make it easier to run a testing loop with `rerun -- bundle exec rake` or similar.

Reply:
This one comes from here: https://github.com/brendon/acts_as_list/commit/a070fad6bf808e89cb86ec69ea227ec03a3a30fe

There is a multi-threaded test routine in the suite that tries to generate a real race condition and I think this change was required in order to get that to happen (otherwise the database is single-threaded). That's my understanding at least :D

There's more detail here: https://github.com/brendon/acts_as_list/commit/79f56a0e09d74744887fc3ed8843481a4b449855

I'm happy to look at removing it provided we don't create a regression in the testing :)